src/lib/Libtpp/tpp_internal.h

```diff
@@ -508,6 +508,17 @@ int tpp_add_fd(int, int, int);
 int tpp_del_fd(int, int);
 int tpp_mod_fd(int, int, int);
 
+/* a new mutex introduced to prevent inheriting lock from tpp thread
+ * from getaddrinfo(nslookup) during fork for periodic hook
+ * set handlers using pthread_atfork.
+ */
+pthread_mutex_t tpp_nslookup_mutex;
+int tpp_nslookup_mutex_lock();
```

Comment:
u can use the tpp_lock() tpp_unlock() routines, no need to add another pair that does exactly the same